### PR TITLE
Match/improve ``xl`` files

### PR DESCRIPTION
**Pull request by Yanis42**
the only non-matchings in these files are ``xlHeapSetup`` and ``xlHeapCompact`` from ``xlHeap.c``, some of the function names are renamed according to what I could find in the gamecube debug stuff (for instance ``xlTextConcat`` became ``xlTextAppend``)

improved:
- ``emulator/xlFile.c`` some of the functions this had originally were moved to ``xlFileRVL.c`` since that's where they belong
- ``emulator/xlHeap.c`` 
- ``emulator/xlList.c``
- ``emulator/xlObject.c``

matched:
- ``emulator/xlCoreRVL.c``
- ``emulator/xlFileRVL.c``
- ``emulator/xlPostRVL.c``
- ``emulator/xlText.c``

btw let me know if I should split the changes with the new matches

### CODE REVIEW

**Comment by Yanis42** [comment]
also if you wonder why only xlList have ``-ipa file`` as an extra flag this is only to fix some inline oddities I encountered, I chose not to apply this flag on the other emu files because it caused weird behaviors sometimes (idr what specifically though)

File: src/emulator/xlHeap.c [hunk 1]
```diff
@@ -1,590 +1,645 @@
...
+    return true;
+}
 
-            if(r0 < size) {
-                r0 = size;
+// https://decomp.me/scratch/2K9NF
+#ifndef NON_MATCHING
+#pragma GLOBAL_ASM("asm/non_matchings/virtual_console/xlHeap/xlHeapCompact.s")
+#else
+bool xlHeapCompact(s32 iHeap) {
```
Comment (cadmic) [inline src/emulator/xlHeap.c:256]: matched: https://decomp.me/scratch/I6co7

changed `pBlockNext = pBlock + nBlockSize + 1` to `pBlockNext = pBlock + 1 + nBlockSize` lol

File: src/emulator/xlHeap.c [hunk 2]
```diff
@@ -1,590 +1,645 @@
...
+    return true;
 }
 
-#ifdef NON_MATCHING
+bool xlHeapGetFree(s32* pnFreeBytes) { return __xlHeapGetFree(0, pnFreeBytes); }
+
+#ifndef NON_MATCHING
+#pragma GLOBAL_ASM("asm/non_matchings/virtual_console/xlHeap/xlHeapSetup.s")
+#else
 s32 xlHeapSetup(void) {
```
Comment (cadmic) [inline src/emulator/xlHeap.c:643]: matched: https://decomp.me/scratch/wrF1B

File: src/emulator/xlCoreRVL.c [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+GXRenderModeObj* rmode;
+
+//! TODO: confirm these
+CNTFileInfo gCNTFileInfo;
+UnknownContentStruct gUnkContent;
+
+//! TODO: find these function names
+u32 fn_800B6F8C();
+
+#define EMU_ROUND_UP(v, x) (((v) + ((x) - 1)) & ~((x) - 1))
```
[the remark this answers is not in the document]
Comment (Yanis42) [reply src/emulator/xlCoreRVL.c:26]: ``ROUND_UP`` seems to work

File: src/emulator/xlHeap.c (hunk 1 above)
Comment (Yanis42) [reply src/emulator/xlHeap.c:256]: I hate that it was this simple lol, thanks for matching this 🎉

File: src/emulator/xlCoreRVL.c [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    while (SCCheckStatus() == 1 && OS_TICKS_TO_MSEC(OSGetTick() - nTickLast) < 3000) {}
+
+    if (mode != NULL) {
+        rmode = mode;
+        return;
+    }
+
+    switch (VIGetTvFormat()) {
+        case VI_NTSC:
+            rmode = VIGetDTVStatus() && ((fn_800B6F8C() & 0xFF) == 1) ? &GXNtsc480Prog : &GXNtsc480IntDf;
```
[the remark this answers is not in the document]
Comment (Yanis42) [reply src/emulator/xlCoreRVL.c]: I had to keep the ``== 1`` to make it match

File: src/emulator/xlHeap.c [hunk 5]
```diff
@@ -28,6 +29,7 @@ u32* gnHeapOS[HEAP_COUNT];
 //! TODO: these need better names
 #define CHKSUM_HI(v) ((u32)((v) >> 26))
 #define CHKSUM_LO(v) ((u32)((v) & 0x3F))
+#define HEAP_ALIGN(v, a) (((u32)(v) + a - 1) & ~(a - 1))
```
Comment (cadmic) [inline src/emulator/xlHeap.c:32]: sorry, missed this. Does ROUND_UP work instead here too?

Comment (Yanis42) [reply src/emulator/xlHeap.c:32]: yeah it works, I just had to add a cast where it's used

**Review by cadmic — APPROVED** [review]

**Pull request merged**